src/sage/graphs/matching_covered_graph.py

```diff
@@ -1141,29 +1141,24 @@ def add_edge(self, u, v=None, label=None):
             if u == v:
                 raise ValueError('loops are not allowed in '
                                  'matching covered graphs')
-
-            # TODO: A ligher incremental method to check whether the new graph
-            # is matching covered instead of creating a new graph and checking
-
-            G = Graph(self, multiedges=self.allows_multiple_edges())
-            G.add_edge(u, v, label=label)
-
-            try:
-                self.__init__(data=G, matching=self.get_matching())
-
-            except Exception:
-                raise ValueError('the graph obtained after the addition of '
+           
+            # If (u, v, label) is a multiple edge/ an existing edge
+            if self.has_edge(u, v):
+                self._backend.add_edge(u, v, label, self._directed)
+                return
+            # Check if there exists an M-alternating odd uv path starting and
+            # ending with edges in self._matching
+            from sage.graphs.matching import M_alternating_even_mark
+            w = next((b if a == u else a) for a, b, *_ in self.get_matching() if u in (a, b))
+            if v in M_alternating_even_mark(self, w, self.get_matching()):
```

Comment:
```suggestion
            w = next((b if a == u else a) for a, b, *_ in self.get_matching() if u in (a, b))

            if v in M_alternating_even_mark(self, w, self.get_matching()):
```
Add an empty line in between.